kubernetes/linux/main.sh

```diff
@@ -1,5 +1,44 @@
 #!/bin/bash
 
+# check to see if Ikey is properly formatted an instrumentation key
+if [[ $APPLICATIONINSIGHTS_AUTH =~ ^[A-Za-z0-9=]+$ ]]  # regex for base64 encoded data
+then
+      # base64 encoded ikey was passed, decode ikey and write to environment variable
+      export APPLICATIONINSIGHTS_AUTH=$(echo $APPLICATIONINSIGHTS_AUTH )
+else
+      # need to fetch ikey from storage account
+      for BACKOFF in {0..5}; do
+            KEY=$(curl -sS $APPLICATIONINSIGHTS_AUTH )
```

Comment:
Why cant  we introduce $APPLICATIONINSIGHTS_AUTH_URL or something easy to understand (IKEY vs URL)and invoke the logic to set APPLICATIONINSIGHTS_AUTH value fetched from SA account and this way you will be covered all the clouds and not touching any existing code? For example, if ($APPLICATIONINSIGHTS_AUTH_URL) { fetch "aikey"}?

Reply:
Good idea, I made this change. Now if there is a custom IKey then APPLICATIONINSIGHTS_AUTH_URL will be set with the URL